### PR TITLE
[docs-only] first iteration for ocis simple guide

### PR DESCRIPTION
**Pull request by micbar**
## Description
Add first guide chapter

## Related Issue
-  Needs https://github.com/owncloud/owncloud.github.io/pull/66 to be merged first

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: docs/ocis/guides/ocis-remote-docker.md [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+For oCIS deployments you often need multiple services. These services need to share resources like volumes and networks. If you do not use any orchestration tool, you would end up writing bash scripts to create and update containers and volumes and connect them via networks. This is what orchestration tools like docker compose can do for you. You define a service mesh using .yaml files and the tool tries to run and maintain that. You gain more value and a version history by using a version control system. Your deployment configuration is fully written down as a spec and you will never touch any system directly and change the config manually.
+
+## Kubernetes
+
+Containers are also used in [kubernetes](https://kubernetes.io/). Kubernetes is part of a huge ecosystem and is founded on best-of-breed practises to orchestrate large scale container applications and services.
+
+## oCIS and Containers
+
+oCIS was developed as microservices. We do not scale the whole system as a monolith but we scale the individual services.
+
```
Comment (butonic) [inline docs/ocis/guides/ocis-remote-docker.md:49]: also true ... but irrelevant for a local single node deployment
```suggestion
```

File: docs/ocis/guides/_index.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+## Step-by-step Guides
+
+These guides should extend the generic documentation and help admins to learn Infinite Scale through digestable "hands-on" sessions.
+
+### Available Guides
+
+oCIS deployments are super simple, yet there are many configurations possible for advanced setups.
+
+- {{< icon "docker" >}} [Discover oCIS with Docker]({{< ref "ocis-remote-docker" >}}) - configure domain, certificates and port
```
Comment (butonic) [inline docs/ocis/guides/_index.md:21]: hm, the file is called ocis-remote-docker, but actually describes a containerized localhost deployment ...

The only mention of domain in the actual guide is in the docker-compose docs
```suggestion
- {{< icon "docker" >}} [Discover oCIS with Docker]({{< ref "ocis-local-docker" >}}) - localhost deployment with secret initialization and persistence of config and data
```

**Review by butonic — DISMISSED** [review]

File: docs/ocis/guides/ocis-remote-docker.md [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+{{< /columns >}}
+
+## Docker compose
+
+For oCIS deployments you often need multiple services. These services need to share resources like volumes and networks. If you do not use any orchestration tool, you would end up writing bash scripts to create and update containers and volumes and connect them via networks. This is what orchestration tools like docker compose can do for you. You define a service mesh using .yaml files and the tool tries to run and maintain that. You gain more value and a version history by using a version control system. Your deployment configuration is fully written down as a spec and you will never touch any system directly and change the config manually.
+
+## Kubernetes
+
+Containers are also used in [kubernetes](https://kubernetes.io/). Kubernetes is part of a huge ecosystem and is founded on best-of-breed practises to orchestrate large scale container applications and services.
+
```
[the remark this answers is not in the document]
Comment (dragotin) [reply docs/ocis/guides/ocis-remote-docker.md:45]: agree here.

File: docs/ocis/guides/ocis-remote-docker.md [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+## oCIS and Containers
+
+oCIS was developed as microservices. We do not scale the whole system as a monolith but we scale the individual services.
+
+## Start oCIS with docker compose
+
+### Use the following skeleton as a docker-compose.yml
```
Comment (dragotin) [inline docs/ocis/guides/ocis-remote-docker.md:52]: Caption should be short, such as "Skeleton for docker-compose.yml" and use another sentence in the text to explain.

File: docs/ocis/guides/ocis-remote-docker.md [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+      OCIS_INSECURE: "true"
+
+      # OCIS_URL: the external domain / ip address of oCIS (with protocol, must always be https)
+      OCIS_URL: "https://localhost:9200"
+
+      # OCIS_LOG_LEVEL: error / info / ... / debug
+      OCIS_LOG_LEVEL: info
+```
+
+### Run ocis init to create a config
```
Comment (dragotin) [inline docs/ocis/guides/ocis-remote-docker.md:79]: Same lengthy caption here.

File: docs/ocis/guides/ocis-remote-docker.md [hunk 6]
```diff
@@ -0,0 +1,343 @@
...
+
+**docker compose exec \<service name\> \<command\>**\
+This command is handy to run specific commands inside your service. Try `docker compose exec ocis ocis version`.
+{{< /hint >}}
+
+### Persist data, restart and logging
+
+The key to a successful container setup is the persistance of the application data to make the data survive a re-boot. Docker normally uses [volumes](https://docs.docker.com/storage/volumes/) for this purpose. A volume can either be a "named volume" which are completely managed by docker and have many advantages (see the linked docker documentation), or "bind mounts" which are uing the directory structure and OS of the host system. In our example we already use a bind mount for the config file. We will now add a named volume for the oCIS data directory.
+
+Add a named volume.
```
Comment (dragotin) [inline docs/ocis/guides/ocis-remote-docker.md:222]: This is either a caption, or superflous?

File: docs/ocis/guides/ocis-remote-docker.md [hunk 7]
```diff
@@ -0,0 +1,343 @@
...
+    logging:
+      driver: local
+    # otherwise you could specify log rotation exlicitely
+    #  driver: "json-file" # this is the default driver
+    #  options:
+    #    max-size: "200k" # limit the size of the log file
+    #    max-file: "10" # limit the count of the log files
+```
+
+Apply your changes! Remember? Just run `docker compose up -d` again.
```
Comment (dragotin) [inline docs/ocis/guides/ocis-remote-docker.md:269]: Remove the "Remember?"

File: docs/ocis/guides/ocis-remote-docker.md [hunk 8]
```diff
@@ -0,0 +1,343 @@
...
+Now you have an oCIS which will survive reboots, restart automatically and has log rotation by default.
+
+Access the logs via `docker compose logs -f` and do some actions in the frontend to see the effect. Create data by uploading files and adding more users. Then run `docker compose down` to shut oCIS down. Start it again `docker compose up -d`, log in again and check that your data has survived the reboot.
+
+### Pin the oCIS version
+
+Last but not least, it is never a good idea to use the `latest` docker tag. Pin your container image to a released version.
+
+```yaml
+    image: owncloud/ocis:2.0.0-beta2
```
Comment (dragotin) [inline docs/ocis/guides/ocis-remote-docker.md:280]: you could go to beta-3 meanwhile ;-)

**Review by dragotin — COMMENTED** [review]
nice!!

File: docs/ocis/guides/ocis-remote-docker.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (micbar) [reply docs/ocis/guides/ocis-remote-docker.md:45]: moved it to another section "ocis and containers" outside of this guide

File: docs/ocis/guides/ocis-remote-docker.md (hunk 1 above)
Comment (micbar) [reply docs/ocis/guides/ocis-remote-docker.md:49]: moved it out

File: docs/ocis/guides/ocis-remote-docker.md (hunk 8 above)
Comment (micbar) [reply docs/ocis/guides/ocis-remote-docker.md:280]: i will use a digest

**Review by dragotin — APPROVED** [review]
:+1:

**Pull request merged**